app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -258,6 +258,18 @@ class MessagingManager @Inject constructor(
             NotificationItem(0, now, jsonData[MESSAGE].toString(), jsonObject.toString(), source)
         notificationDao.add(notificationRow)
 
+        mainScope.launch {
+            if (integrationUseCase.getNotificationReceivedEvent()) {
+                try {
+                    integrationUseCase.fireEvent("mobile_app_notification_received", jsonData)
+                } catch (e: Exception) {
+                    Log.e(TAG, "Unable to send notification received event", e)
+                    Toast.makeText(context, commonR.string.notification_received_error, Toast.LENGTH_LONG).show()
```

Comment:
I'm not a big fan of using Toast messages:
 - They lack context. It's hard to know which notification this message belongs to, especially if the notification was a command, triggered by an automation and/or multiple notifications were delivered shortly after another (for example due to doze). This is only made worse by the duration and how Toasts are queued to display.
 - They are not actionable. So sending the event failed, what should I do now? You can't manually send it again. Manually doing something when there's no confirmation event defeats the purpose of the event, that should be handled by the automation.

As a result, the only result of the message is that you get frustrated and/or have to open the app to find out more and do something about it.

I'd say simply printing the error in the logs is enough. If you feel like a more visible error is important, maybe save the result as a boolean in the database with the specific notification and show it in the notification history details?

Reply:
Ok I can remove the toast message, we should probably do the same in the other PR for the clear event failure in this case.

Reply:
Are you referring to #2576? These arguments still apply but are a little weaker in that case because it is the result of a direct action by the user (but still very much in favor of removing the toast there! ).

Reply:
Ok I think it makes sense, some of our Toast messages were from before the in app log viewer. Now that we've had it for sometime it's an ok place to tell users where to go.